inginious/agent/docker_agent/__init__.py

```diff
@@ -437,13 +443,26 @@ async def create_student_container(self, parent_info, socket_id, environment_nam
             environment = self._containers[environment_type][environment_name]["id"]
             runtime = self._containers[environment_type][environment_name]["runtime"]
 
+            if ssh:
```

Comment:
As this is simple, you can save 3 lines by writing : ``ports_needed = [22] if ssh else []`` :)